crates/sui-framework/packages/sui-system/sources/voting_power.move

```diff
@@ -60,17 +60,17 @@ public(package) fun set_voting_power(validators: &mut vector<Validator>) {
 fun init_voting_power_info(
     validators: &vector<Validator>,
     threshold: u64,
+    total_stake: u64,
 ): (vector<VotingPowerInfoV2>, u64) {
-    let total_stake = total_stake(validators);
     let mut i = 0;
     let len = validators.length();
     let mut total_power = 0;
     let mut result = vector[];
     while (i < len) {
         let validator = &validators[i];
         let stake = validator.total_stake();
-        let adjusted_stake = stake as u128 * (TOTAL_VOTING_POWER as u128) / (total_stake as u128);
-        let voting_power = (adjusted_stake as u64).min(threshold);
+        let adjusted_stake = derive_voting_power(stake, total_stake);
```

Comment:
Variable name a bit confusing here (yes, I wrote this bad code not you--apologies) since no adjustment is going on here--I'd probably suggest `let voting_power = derive_voting_power(stake, total_stake).min(threshold)`. But also fine to land as is.

Reply:
Done. Thank you!